### PR TITLE
Add EmoteOnDamage comp/system for zombies

### PR DESCRIPTION
**Pull request by 0x6273**
Gets rid of `ActiveZombieComponent`, adds new `EmoteOnDamageComponent` and system that does the same thing but more reusable.

It's similar to AutoEmote but I decided not to support per-emote chance and cooldown values on this one to keep it simple.

Also, this fixes a bug where if a mob becomes a zombie some other way than changing mobstate to critical (such as admin antag ctrl, or the "turn into zombie" action) they would not get all of the zombie behavior.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
Do the emotes still play on damage if the entity is incapacitated?

**Comment by 0x6273** [comment]
> Do the emotes still play on damage if the entity is incapacitated?

I don't think I've changed anything related to this. They should play only when mobstate is `MobState.Alive`.

**Comment by brainfood1183** [comment]
we need a system that handles infection on damage like how zombies infect crew and turn them into zombies so we can apply this to other forms of infection from "conversion" monsters.

File: Content.Server/Chat/Systems/EmoteOnDamageSystem.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            return;
+
+        if (emoteOnDamage.Emotes.Count == 0)
+            return;
+
+        if (!_random.Prob(emoteOnDamage.EmoteChance))
+            return;
+
+        var emote = _random.Pick(emoteOnDamage.Emotes);
+        _chatSystem.TryEmoteWithoutChat(uid, emote);
```
Comment (AJCM-git) [inline Content.Server/Chat/Systems/EmoteOnDamageSystem.cs:45]: make WithoutChat opt-out with a field in the component

**Review by AJCM-git — CHANGES_REQUESTED** [review]
looks good (a video in the pr body would be appreciated)

File: Content.Server/Chat/EmoteOnDamageComponent.cs [hunk 2]
```diff
@@ -24,6 +24,19 @@ public sealed class EmoteOnDamageComponent : Component
...
+    /// <summary>
+    [DataField("withChat"), ViewVariables(VVAccess.ReadWrite)]
+    public bool WithChat = false;
+
+    /// <summary>
+    /// Hide the chat message from the chat window, only showing the popup.
+    /// This does nothing if WithChat is false.
+    /// <summary>
+    [DataField("hiddenFromChatWindow")]
+    public bool HiddenFromChatWindow = false;
```
Comment (AJCM-git) [inline Content.Server/Chat/EmoteOnDamageComponent.cs:38]: arent these the same or am i missing something? (also dont forget the video in the pr body)

**Comment by AJCM-git** [comment]
@0x6273 you coming back to this?

**Comment by 0x6273** [comment]
> @0x6273 you coming back to this?

It's ready for review/merge. There's no video because it doesn't change anything that's visible to players.

**Pull request merged**